tests/prophet/test_prophet_model_export.py

```diff
@@ -0,0 +1,455 @@
+            registered_model_name="ProphetModel1",
+        )
+        model_uri = f"runs:/{mlflow.active_run().info.run_id}/{artifact_path}"
+        mlflow.register_model.assert_called_once_with(
+            model_uri, "ProphetModel1", await_registration_for=DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+        )
+
+
+@pytest.mark.large
+def test_log_model_no_registered_model_name(prophet_model):
+    artifact_path = "prophet"
+    register_model_patch = mock.patch("mlflow.register_model")
+    with mlflow.start_run(), register_model_patch, TempDir(chdr=True, remove_on_exit=True) as tmp:
+        conda_env = os.path.join(tmp.path(), "conda_env.yaml")
+        _mlflow_conda_env(conda_env, additional_pip_deps=["pystan", "prophet"])
+        mlflow.prophet.log_model(
+            pr_model=prophet_model.model, artifact_path=artifact_path, conda_env=conda_env,
+        )
+        mlflow.register_model.assert_not_called()
+
+
+@pytest.mark.large
+def test_model_save_persists_specified_conda_env_in_mlflow_model_directory(
+    prophet_model, model_path, prophet_custom_env
+):
+    mlflow.prophet.save_model(
+        pr_model=prophet_model.model, path=model_path, conda_env=prophet_custom_env
+    )
+
+    pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
+    saved_conda_env_path = os.path.join(model_path, pyfunc_conf[pyfunc.ENV])
+    assert os.path.exists(saved_conda_env_path)
+    assert saved_conda_env_path != prophet_custom_env
+
+    with open(prophet_custom_env, "r") as f:
+        prophet_custom_env_parsed = yaml.safe_load(f)
+    with open(saved_conda_env_path, "r") as f:
+        saved_conda_env_parsed = yaml.safe_load(f)
+    assert saved_conda_env_parsed == prophet_custom_env_parsed
+
+
+@pytest.mark.large
+def test_model_save_persists_requirements_in_mlflow_model_directory(
+    prophet_model, model_path, prophet_custom_env
+):
+    mlflow.prophet.save_model(
+        pr_model=prophet_model.model, path=model_path, conda_env=prophet_custom_env
+    )
+
+    saved_pip_req_path = os.path.join(model_path, "requirements.txt")
+    _compare_conda_env_requirements(prophet_custom_env, saved_pip_req_path)
+
+
+@pytest.mark.large
+def test_log_model_with_pip_requirements(prophet_model, tmpdir):
+    req_file = tmpdir.join("requirements.txt")
+    req_file.write("a")
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(prophet_model.model, "model", pip_requirements=req_file.strpath)
+        _assert_pip_requirements(mlflow.get_artifact_uri("model"), ["mlflow", "a"], strict=True)
+
+    # List of requirements
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(
+            prophet_model.model, "model", pip_requirements=[f"-r {req_file.strpath}", "b"]
+        )
+        _assert_pip_requirements(
+            mlflow.get_artifact_uri("model"), ["mlflow", "a", "b"], strict=True
+        )
+
+    # Constraints file
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(
```

Comment:
Looks like we're accidentally logging the models with `mlflow.sklearnn.log_model()` here, rather than `mlflow.prophet.log_model()`. Seems like pickling works here though :D

Reply:
wow that was a big oversight. Of note: per their docs (and my testing) pickling works just fine so long as the version of pystan is consistent with the underlying built solvers - but if pystan version is different it blows up. However, if the ser/deser that's used with their approach is used, then the environment won't matter and reproducibility can work fine. Seems like the root of all of these woes is in pystan from what I can tell.